src/components/map/Map.jsx

```diff
@@ -34,20 +39,22 @@ function Map({ mapData, zoomControl, additionalClass = "", withIndex }) {
         attribution='&copy; <a href="https://www.openstreetmap.org/copyright">OpenStreetMap</a> contributors'
         url="https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png"
       />
-      {mapData.map(({ latitude, longitude }, index) => (
-        <Marker
-          key={latitude}
-          position={[latitude, longitude]}
-          icon={L.divIcon({
-            html: getHtmlPin(avoidZeroIndexingPoints(index)),
-            // The empty string classname below seems like something to remove, but if I remove it, a little square
-            // appears in the map... Not sure why
-            className: "",
-            iconSize: [24, 24], // If icon size is changed, it should equally be changed in the map-wrapper.css
-            iconAnchor: [12, 24], // 12 centers it, 24 makes the pointy end point on the right coordinates
-          })}
-        />
-      ))}
+      {lat &&
```

Comment:
Wouldn't it make more sense to have a separate `Marker` (outside the `map` loop) to show the current position? This would probably make it easier to make it visually different from the route points.